embedded-fatfs/src/dir.rs

```diff
@@ -194,6 +194,25 @@ impl<'a, IO: ReadWriteSeek, TP: TimeProvider, OCC: OemCpConverter> Dir<'a, IO, T
         }
     }
 
+    /// Check to see if a file or directory with the given name exists
+    pub async fn exists(&self, mut path: &str, is_dir: Option<bool>) -> Result<bool, Error<IO::Error>> {
```

Comment:
`is_dir: Option<bool>` in the API feels a bit weird, I don't think it's needed.

IIRC, if `is_dir` is `None`, `find_entry` will return the entry whether it's a file or a dir. Could you check this, and also add a test case?